node/peer.js

```diff
@@ -45,11 +45,22 @@ function TChannelPeer(channel, hostPort, options) {
     self.state = null; // TODO
     self.connections = [];
     self.random = self.channel.random;
+
+    self.stateOptions = {
+        timers: self.channel.timers,
+        random: self.channel.random,
+        period: self.options.period,
+        maxErrorRate: self.options.maxErrorRate,
+        probation: self.options.probation,
+        stateMachine: self,
```

Comment:
Intent is to reuse for circuit. In the circuit case, the stateMachine is the circuit, and the nextHandler.shouldRequest just returns 1.